packages/models/src/tenant/protobufConverterFromV2.ts

```diff
@@ -0,0 +1,192 @@
+import { createHash } from "crypto";
+import { match } from "ts-pattern";
+import { unsafeBrandId } from "../brandedIds.js";
+import { genericError } from "../errors.js";
+import {
+  TenantKindV2,
+  TenantMailKindV2,
+  TenantMailV2,
+  TenantFeatureV2,
+  TenantVerifierV2,
+  TenantRevokerV2,
+  TenantAttributeV2,
+  TenantV2,
+  TenantUnitTypeV2,
+} from "../gen/v2/tenant/tenant.js";
+import {
+  TenantKind,
+  tenantKind,
+  TenantMailKind,
+  tenantMailKind,
+  TenantMail,
+  TenantFeatureCertifier,
+  TenantVerifier,
+  TenantRevoker,
+  TenantAttribute,
+  Tenant,
+  ExternalId,
+  tenantAttributeType,
+  TenantUnitType,
+  tenantUnitType,
+} from "./tenant.js";
+
+export const fromTenantKindV2 = (input: TenantKindV2): TenantKind => {
+  switch (input) {
+    case TenantKindV2.GSP:
+      return tenantKind.GSP;
+    case TenantKindV2.PA:
+      return tenantKind.PA;
+    case TenantKindV2.PRIVATE:
+      return tenantKind.PRIVATE;
+    case TenantKindV2.UNSPECIFIED$:
+      throw new Error("Unspecified tenant kind");
+  }
+};
+
+export const fromTenantMailKindV2 = (
+  input: TenantMailKindV2
+): TenantMailKind => {
+  switch (input) {
+    case TenantMailKindV2.CONTACT_EMAIL:
+      return tenantMailKind.ContactEmail;
+    case TenantMailKindV2.DIGITAL_ADDRESS:
+      return tenantMailKind.DigitalAddress;
+    case TenantMailKindV2.UNSPECIFIED$:
+      throw new Error("Unspecified tenant mail kind");
+  }
+};
+
+export const fromTenantMailV2 = (input: TenantMailV2): TenantMail => ({
+  ...input,
+  id: input.id
+    ? input.id
+    : createHash("sha256").update(input.address).digest("hex"),
```

Comment:
This could be more readable
```suggestion
  id: input.id ?? createHash("sha256").update(input.address).digest("hex"),
```

Reply:
Updated, thanks